File: Book/Fortran/Fig_6.4_wrongPrivate.f90
! sample compile command to generate *.o object file:
!     gfortran -fopenmp -c Fig_6.4_wrongPrivate.f90

program wrong
   integer :: tmp
   tmp = 0

!$omp parallel do private(tmp)
   do j = 1, 1000
      tmp = tmp +j
   enddo
!$omp end parallel do

   print *, tmp   ! tmp is 0 here
end program wrong

File: Book/C/Fig_6.4_wrongPrivate.c
// sample compile command: "gcc -fopenmp -c Fig_6.4_wrongPrivate.c" to generate *.o object file

#include <stdio.h>
void wrong() 
{ 
   int tmp = 0;
   #pragma omp parallel for private(tmp) 
      for (int j = 0; j < 1000; j++)
         tmp += j;   
      printf("%d\n", tmp);  //tmp is 0 here
}
